compiler/rustc_typeck/src/check/method/suggest.rs

```diff
@@ -478,6 +478,16 @@ impl<'a, 'tcx> FnCtxt<'a, 'tcx> {
                 let mut label_span_not_found = || {
                     if unsatisfied_predicates.is_empty() {
                         err.span_label(span, format!("{item_kind} not found in `{ty_str}`"));
+                        if matches!(actual.kind(), ty::Ref(_, ty, _) if ty.is_str())
+                            && item_name.name == sym::iter
+                        {
+                            err.span_suggestion(
+                                item_name.span,
+                                "did you mean",
+                                String::from("chars"),
+                                Applicability::MaybeIncorrect,
```

Comment:
The likelihood of this being correct is quite high, given the checks you have above.

```suggestion
                                Applicability::MachineApplicable,
```